### PR TITLE
Support torch expand with const

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
Add torch.Tensor.expand to TPC.v4.
Fix const quantization for torch expand & gather.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/target_platform_capabilities/tpc_models/imx500_tpc/v4/tpc_pytorch.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 from torch import add, sub, mul, div, divide, flatten, reshape, split, unsqueeze, dropout, sigmoid, tanh, \
     chunk, unbind, topk, gather, equal, transpose, permute, argmax, squeeze, multiply, subtract
 from torch.nn import Conv2d, Linear, ConvTranspose2d, MaxPool2d
-from torch.nn import Dropout, Flatten, Hardtanh, Identity
+from torch.nn import Dropout, Flatten, Hardtanh
```
Comment (lior-dikstein) [inline model_compression_toolkit/target_platform_capabilities/tpc_models/imx500_tpc/v4/tpc_pytorch.py:22]: Did you erase it because it is not used?

File: tests/pytorch_tests/model_tests/feature_models/const_quantization_test.py [hunk 2]
```diff
@@ -196,3 +198,78 @@ def compare(self, quantized_model, float_model, input_x=None, quantization_info=
...
+                                                   weights_quantization_method=tp.QuantizationMethod.POWER_OF_TWO))
+        const_configuration_options = tp.QuantizationConfigOptions([const_config])
+
+        tp_model = tp.TargetPlatformModel(default_configuration_options)
+        with tp_model:
+            tp.OperatorsSet("WeightQuant", const_configuration_options)
+
+        tpc = tp.TargetPlatformCapabilities(tp_model)
+        with tpc:
+            # No need to quantize Flatten and Dropout layers
```
Comment (lior-dikstein) [inline tests/pytorch_tests/model_tests/feature_models/const_quantization_test.py]: I don't understand the comment

**Review by lior-dikstein — APPROVED** [review]
